pkg/cloud/externalconn/connectionpb/connection.proto

```diff
@@ -18,23 +18,35 @@ import "cloud/cloudpb/external_storage.proto";
 enum ConnectionProvider {
   Unknown = 0;
   nodelocal = 1;
+  gs_kms = 2;
 }
 
 enum ConnectionType {
   option (gogoproto.goproto_enum_prefix) = false;
 
   UNSPECIFIED = 0 [(gogoproto.enumvalue_customname) = "TypeUnspecified"];
   STORAGE = 1 [(gogoproto.enumvalue_customname) = "TypeStorage"];
+  KMS = 2 [(gogoproto.enumvalue_customname) = "TypeKMS"];
 }
 
 message ConnectionDetails {
   ConnectionProvider provider = 1;
 
   oneof details {
     NodelocalConnectionDetails nodelocal = 2;
+    GCSKMSConnectionDetails gcs_kms = 3 [(gogoproto.customname) = "GCSKMS"];
   }
 }
 
 message NodelocalConnectionDetails {
   cloudpb.LocalFileConfig cfg = 1 [(gogoproto.nullable) = false];
 }
+
+message GCSKMSConnectionDetails {
```

Comment:
Another thought that crossed my mind while working on kafka support is if we should even be disassembling this URI only to put it back together when it is used. This becomes quite cumbersome when you start having 10s of params like kafka. Why not just stick the URI string in this proto after performing the relevant validation during `CREATE EXTERNAL CONNECTION`. Thoughts @miretskiy @benbardin